integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/adapters.py

```diff
@@ -252,10 +254,18 @@ def _extract_messages_from_response(self, response_body: Dict[str, Any]) -> List
         """
         messages: List[ChatMessage] = []
         if response_body.get("type") == "message":
-            for content in response_body["content"]:
-                if content.get("type") == "text":
-                    meta = {k: v for k, v in response_body.items() if k not in ["type", "content", "role"]}
-                    messages.append(ChatMessage.from_assistant(content["text"], meta=meta))
+            if response_body.get("stop_reason") == "tool_use":  # If `tool_use` we only keep the tool_use content
+                for content in response_body["content"]:
+                    if content.get("type") == "tool_use":
+                        meta = {k: v for k, v in response_body.items() if k not in ["type", "content", "role"]}
+                        json_answer = json.dumps(content)
+                        messages.append(ChatMessage.from_assistant(json_answer, meta=meta))
+            else:  # For other stop_reason, return all text content
+                for content in response_body["content"]:
+                    if content.get("type") == "text":
+                        meta = {k: v for k, v in response_body.items() if k not in ["type", "content", "role"]}
+                        messages.append(ChatMessage.from_assistant(content["text"], meta=meta))
+
```

Comment:
Seems a bit complex, no?

What about smth like:
```
if response_body.get("type") == "message":
    meta = {k: v for k, v in response_body.items() if k not in ["type", "content", "role"]}
    for content in response_body["content"]:
        text = content.get("text") or json.dumps(content.get("tool_use"))
        if text:
            messages.append(ChatMessage.from_assistant(text, meta=meta))

return messages
```

Reply:
The problem of this is that as I explained when doing a tool use sometimes the LLM answer two replies leading to issues.

> Either the model answer only ONE reply that is the tool_use type like {'type': 'tool_use', 'id': 'toolu_bdrk_01NcAB4KKhQzdV6eX1ji1Hpk', 'name': 'top_song', 'input': {'sign': 'WZPZ'}} => TEST IS PASSING

> Or the model produce two replies:

> One text type answer such as {'type': 'text', 'text': 'Here is the most popular song played on radio station WZPZ:'}
> One tool_use type answer as before: {'type': 'tool_use', 'id': 'toolu_bdrk_01NcAB4KKhQzdV6eX1ji1Hpk', 'name': 'top_song', 'input': {'sign': 'WZPZ'}}

> But in this case only one instance of meta data is present (prompt/completion/input/ouput tokens).

The two answer situation leads to failing test (flacky) and also poor developper experience because now for two replies you only have one set of meta-data and you can't predict if the answer is valid JSON or just a 'pre-message' before the valid JSON for tool-use.

That's why my original code, in the case of a tool-use stop-sequence allow to discard the text answer before the actual tool-use JSON.

Reply:
> Thanks for your patience. I think I know what is going on. You are getting the "thinking" [message](https://docs.anthropic.com/en/docs/build-with-claude/tool-use#chain-of-thought). We dealt with this by having a boolean switch in Anthropic to account for this thinking message. See #790 for more details.
> 
> What do you think we add the same init parameter here? Open to suggestions...

I think it's exactly what it is
 I guess it will need a bit more time for me to finish this PR then to add this new feature ^^

Reply:
I'm not sure however where to put this parameters, because the `AmazonBedrockChatGenerator` is a generalist generator for a lot of models, so it's not great to put a specific init parameters for Anthropic model in its init method

Reply:
Hmm, right. So we leave it as-is is (i.e. the current state of PR) is the best option, no?

Reply:
> Hmm, right. So we leave it as-is is (i.e. the current state of PR) is the best option, no?


I guess yes, I don't see in which case somebody would need these thinking messages